app/frontend/src/components/forms/manage/ManageFormActions.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div>
+    <span>
+      <ShareForm :formId="form.id" :warning="!hasVersions"/>
+    </span>
+
+    <span v-if="canManageTeam">
+      <v-tooltip bottom>
+        <template #activator="{ on, attrs }">
+          <router-link :to="{ name: 'FormTeams', query: { f: form.id } }">
+            <v-btn class="mx-1" color="primary" icon v-bind="attrs" v-on="on">
+              <v-icon>group</v-icon>
+            </v-btn>
+          </router-link>
+        </template>
+        <span>Team Management</span>
+      </v-tooltip>
+    </span>
+
+    <span v-if="canDeleteForm">
+      <v-tooltip bottom>
+        <template #activator="{ on, attrs }">
+          <v-btn
+            class="mx-1"
+            color="red"
+            @click="showDeleteDialog = true"
+            icon
+            v-bind="attrs"
+            v-on="on"
+          >
+            <v-icon>delete</v-icon>
+          </v-btn>
+        </template>
+        <span>Delete Form</span>
+      </v-tooltip>
+
+      <BaseDialog
+        v-model="showDeleteDialog"
+        type="CONTINUE"
+        @close-dialog="showDeleteDialog = false"
+        @continue-dialog="deleteForm"
+      >
+        <template #title>Confirm Deletion</template>
+        <template #text>
+          Are you sure you wish to delete
+          <strong>{{ form.name }}</strong
+          >? This form will no longer be accessible.
+        </template>
+        <template #button-text-continue>
+          <span>Delete</span>
+        </template>
+      </BaseDialog>
+    </span>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions } from 'vuex';
+
+import { FormPermissions } from '@/utils/constants';
+import ShareForm from '@/components/forms/manage/ShareForm.vue';
+
+export default {
+  name: 'ManageFormActions',
+  components: { ShareForm },
+  data() {
+    return {
+      showDeleteDialog: false,
+    };
+  },
+  computed: {
+    ...mapGetters('form', ['form', 'permissions']),
+    // Permission checks
+    canDeleteForm() {
+      return this.permissions.includes(FormPermissions.FORM_DELETE);
+    },
+    canManageTeam() {
+      return this.permissions.includes(FormPermissions.TEAM_UPDATE);
+    },
+    hasVersions() {
+      return this.form.versions && this.form.versions.length;
```

Comment:
Do we need to also check for `this.form` in the event form is undefined?

Reply:
Will leave for now it since if there's no form the entire page that uses this component will have crashed out anyways.